homeassistant/components/camera/media_source.py

```diff
@@ -71,36 +86,33 @@ async def async_browse_media(
 
         can_stream_hls = "stream" in self.hass.config.components
 
-        # Root. List cameras.
-        component: EntityComponent[Camera] = self.hass.data[DOMAIN]
-        children = []
-        not_shown = 0
-        for camera in component.entities:
+        async def _filter_browsable_camera(camera: Camera) -> BrowseMediaSource | None:
             stream_type = camera.frontend_stream_type
 
             if stream_type is None:
                 content_type = camera.content_type
 
-            elif can_stream_hls and stream_type == StreamType.HLS:
+            elif can_stream_hls:
                 content_type = FORMAT_CONTENT_TYPE[HLS_PROVIDER]
-
+                if stream_type != StreamType.HLS:
+                    source = await camera.stream_source()
+                    if not source:
+                        return None
```

Comment:
Does it work for this to be a single statement? (Given this is 3 levels of nested ifs in an inline function)

```suggestion
                if stream_type != StreamType.HLS and not (source := await camera.stream_source())
                    return None
```

Reply:
I took advantage of the ability to return early in a successful case to refactor a bit differently. Let me know what you think.